.github/workflows/publish-docker-e2e-service.yaml

```diff
@@ -0,0 +1,65 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: publish-docker-e2e-service
+
+on:
+  push:
```

Comment:
`on/push` seems could make image push if someone created a branch in the main repo, like you are doing.

Reply:
> `on/push` seems could make image push if someone created a branch in the main repo, like you are doing.

First, we have a `paths` keyword inside `push` as a filter.

Then, we also have a `if` condition check in the last step to prevent `docker push` from running.

Reply:
Add `branches` is a good practice if you plan to run this workflow in master branch, people, if you want to test this PR, add the branch name of this PR to the `branches` and remove it before ready